### PR TITLE
Update type of `transport` param.

### PR DESCRIPTION
**Pull request by lpender**
This change resolves TS errors for me locally.

See associated commits:

https://github.com/phoenixframework/phoenix/commit/514af3f477ccb88c2a978ea51f1eacf6ed49d902#diff-bae6505f39e00a1c0a8925699e73a03f7680f7e1fff357a782910daf9bee5501

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by princemaple** [comment]
Phoenix js source code now specifies Function. Could you try that?

File: types/phoenix/index.d.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export type ConnectionState = 'connecting' | 'open' | 'closing' | 'closed';
 export interface SocketConnectOption {
   binaryType: BinaryType;
   params: object | (() => object);
-  transport: string;
+  transport: object;
```
Comment (lpender) [inline types/phoenix/index.d.ts:42]:
```suggestion
  transport: Function;
```

**Comment by lpender** [comment]
Hey, thanks. Yes, function works. I updated the PR.

File: types/phoenix/index.d.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export type ConnectionState = 'connecting' | 'open' | 'closing' | 'closed';
 export interface SocketConnectOption {
   binaryType: BinaryType;
   params: object | (() => object);
-  transport: string;
+  transport: Function;
```
Comment (princemaple) [inline types/phoenix/index.d.ts:42]:
```suggestion
  transport: new (endpoint: string) => Object;
```

**Review by princemaple — CHANGES_REQUESTED** [review]
Looks like `Function` made CI unhappy. Could you try this instead?

File: types/phoenix/index.d.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ export type ConnectionState = 'connecting' | 'open' | 'closing' | 'closed';
 export interface SocketConnectOption {
   binaryType: BinaryType;
   params: object | (() => object);
-  transport: string;
+  transport: new (endpoint: string) => Object;
```
Comment (lpender) [inline types/phoenix/index.d.ts:42]:
```suggestion
  transport: new (endpoint: string) => object;
```

**Comment by lpender** [comment]
Hey, thanks for the suggestion @princemaple! The changes removed local errors for me. Would love to get it merged as it's holding up a build/deploy for me.

Thanks

image

**Review by princemaple — APPROVED** [review]
LGTM

**Review by amcasey — APPROVED** [review]

**Comment by lpender** [comment]
Ready to merge

**Pull request merged**